velox/exec/fuzzer/MemoryArbitrationFuzzer.cpp

```diff
@@ -0,0 +1,665 @@
+    core::PlanNodeId scanId;
+    auto plan = PlanWithSplits{
+        PlanBuilder(planNodeIdGenerator)
+            .tableScan(inputRowType)
+            .capturePlanNodeId(scanId)
+            .partialAggregation(groupingKeys, aggregates, {})
+            .intermediateAggregation()
+            .finalAggregation()
+            .planNode(),
+        {{scanId, fromConnectorSplits(splits)}}};
+    plans.push_back(std::move(plan));
+  }
+
+  return plans;
+}
+
+void MemoryArbitrationFuzzer::verify() {
+  const auto outputDirectory = TempDirectoryPath::create();
+  const auto spillDirectory = exec::test::TempDirectoryPath::create();
+  const auto tableScanDir = exec::test::TempDirectoryPath::create();
+
+  std::vector<PlanWithSplits> plans;
+  for (const auto& plan : hashJoinPlans(tableScanDir->getPath())) {
+    plans.push_back(plan);
+  }
+  for (const auto& plan : aggregatePlans(tableScanDir->getPath())) {
+    plans.push_back(plan);
+  }
+
+  SCOPE_EXIT {
+    waitForAllTasksToBeDeleted();
+  };
+
+  const auto startTime = std::chrono::system_clock::now();
+  const auto numThreads = FLAGS_num_batches;
+  std::vector<std::thread> queryThreads;
+  queryThreads.reserve(numThreads);
+  for (int i = 0; i < numThreads; ++i) {
+    queryThreads.emplace_back([&, i]() {
+      while (true) {
+        if (static_cast<std::chrono::duration<double>>(
+                std::chrono::system_clock::now() - startTime)
+                .count() > FLAGS_iteration_duration_sec) {
+          break;
+        }
+
+        try {
+          const auto queryCtx = newQueryCtx(
+              memory::memoryManager(),
+              executor_.get(),
+              FLAGS_arbitrator_capacity);
+          const auto plan = plans.at(randInt(0, plans.size() - 1));
+          AssertQueryBuilder builder(plan.plan);
+          builder.queryCtx(queryCtx);
+          for (const auto& [planNodeId, nodeSplits] : plan.splits) {
+            builder.splits(planNodeId, nodeSplits);
+          }
+
+          if (vectorFuzzer_.coinToss(0.1)) {
+            builder.queryCtx(queryCtx).copyResults(pool_.get());
+          } else {
+            auto res =
+                builder.configs(queryConfigsWithSpill_)
+                    .spillDirectory(
+                        spillDirectory->getPath() + fmt::format("/{}/", i))
+                    .queryCtx(queryCtx)
+                    .copyResults(pool_.get());
+          }
+          ++stats_.success;
+        } catch (const VeloxException& e) {
+          if (e.errorCode() != error_code::kMemCapExceeded.c_str() &&
+              e.errorCode() != error_code::kMemAborted.c_str() &&
+              e.errorCode() != error_code::kMemAllocError.c_str() &&
+              (e.message() != "Aborted for external error")) {
+            std::rethrow_exception(std::current_exception());
+          }
+
+          if (e.message() == "Aborted for external error") {
+            stats_.errors["EXTERNAL_ERROR"]++;
```

Comment:
Let's have abortErrors and oomErrors? Thanks!

Reply:
They are handled in the else branch.
```C++
} else {
            stats_.errors[e.errorCode()]++;
          }
```

Reply:
I mean we just two stats counter? one called oomCount, abortCount? Thanks!